hack/update-openapigen.sh

```diff
@@ -17,15 +17,39 @@
 set -o errexit
 set -o nounset
 
+source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scripts/library.sh
 source $(git rev-parse --show-toplevel)/hack/setup-temporary-gopath.sh
+
+readonly TMP_DIFFROOT="$(mktemp -d ${REPO_ROOT_DIR}/tmpdiffroot.XXXXXX)"
+
+cleanup() {
+  rm -rf "${TMP_DIFFROOT}"
+  shim_gopath_clean
+}
+
+cleanup
+
 shim_gopath
-trap shim_gopath_clean EXIT
+mkdir -p "${TMP_DIFFROOT}"
+
+trap cleanup EXIT
 
 echo "Generating OpenAPI specification ..."
 go run k8s.io/kube-openapi/cmd/openapi-gen \
     --input-dirs ./pkg/apis/pipeline/v1beta1,./pkg/apis/pipeline/pod,./pkg/apis/resource/v1alpha1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1beta1 \
     --output-package ./pkg/apis/pipeline/v1beta1 -o ./ \
-    --go-header-file hack/boilerplate/boilerplate.go.txt >> /dev/null
+    --go-header-file hack/boilerplate/boilerplate.go.txt \
+    -r "${TMP_DIFFROOT}/api-report"
+
+violations=$(diff --changed-group-format='%>' --unchanged-group-format='' "hack/ignored-openapi-violations.list" "${TMP_DIFFROOT}/api-report" || echo "")
```

Comment:
@vdemeester (and also @dibyom, since we'd want the same thing in Triggers once it's calling `openapi-gen`):

So this will diff the list of API violations from running `openapi-gen` against a known list of ignored violations (in `hack/ignored-openapi-violations.list`) and error out if there are any lines in in the report that aren't in the ignore list. I don't love it, but it works and allows us not to rename fields that currently violate `names_match`.